app/workers/vendor_api_request_worker.rb

```diff
@@ -0,0 +1,37 @@
+class VendorAPIRequestWorker
+  include Sidekiq::Worker
+  sidekiq_options retry: 3, queue: :low_priority
+
+  def perform(request_data, response_body, status_code, response_time, created_at)
+    request_headers = request_data['headers']
+    hashed_token = hashed_auth_token(request_headers.delete('HTTP_AUTHORIZATION'))
+
+    VendorAPIRequest.create(
+      request_path: request_data['path'],
+      request_headers: request_headers,
+      request_body: request_data['params'],
+      response_body: response_hash(response_body, status_code),
+      status_code: status_code,
+      response_time: response_time,
+      hashed_token: hashed_token,
```

Comment:
Still a bit unsure about this. Should we look up the provider and save that? Or alternatively a link to the VendorApiToken?

Reply:
:+1: Yeah I think it's possible to look up the api user based on this value. Will add.